### PR TITLE
legion: Update Python dependencies. Fix variant requirements. Remove TLS.

### PR DESCRIPTION
**Pull request by elliottslaughter**
This includes several fixes and quality of life changes for Legion:

 * `+python` requires `python` >= 3.8
 * `+python` depends on `py-cffi` and `py-numpy`
 * `+python` and `+fortran` both require `+bindings`
 * `+python` requires `+shared`
 * `Legion_ENABLE_TLS` was removed in December 2020 and therefore is not present in any version of Legion captured in this package: https://github.com/StanfordLegion/legion/commit/b2a091f39bb1c242ddebbf77dea76911b125107d

~I'm not sure I'm doing the variant requirements correctly. Please let me know if there is a better way to do that.~ (Updated to use `requires` per Slack  discussion.)

CC @rbberger, who might care about some of these changes.

### CODE REVIEW

**Comment by rbberger** [comment]
@elliottslaughter btw, the `python@3` requirement isn't good enough, at least for `legion_prof.py` to work, since it requires support for `from __future__ import annotations`  (https://github.com/StanfordLegion/legion/blame/stable/tools/legion_prof.py#L20). I believe it should be something like `python@3.8:`

**Comment by elliottslaughter** [comment]
@eddy16112, what's the correct minimum Python version for legion_prof.py?

**Comment by eddy16112** [comment]
python annotations requires 3.8

**Comment by elliottslaughter** [comment]
@eddy16112 Is this accurate? https://gitlab.com/StanfordLegion/legion/-/commit/c697fd3345bfe1ee8ceb8a14f1d7d2e9abeb8fd9

**Comment by elliottslaughter** [comment]
I updated the minimum Python version bounds and switched over to the `requires` syntax as suggested in Slack. Please take another look and see if that looks ok.

**Comment by rbberger** [comment]
@elliottslaughter LGTM

**Comment by eddy16112** [comment]
> @eddy16112 Is this accurate? https://gitlab.com/StanfordLegion/legion/-/commit/c697fd3345bfe1ee8ceb8a14f1d7d2e9abeb8fd9

LGTM

**Review by eugeneswalker — APPROVED** [review]

**Comment by eugeneswalker** [comment]
Defer to you as to whether to wait for review from @pmccormick

**Comment by elliottslaughter** [comment]
I checked with @pmccormick offline and I think it's ok to merge this now.

**Pull request merged**
